test/e2e/support/custom.ts

```diff
@@ -862,75 +862,54 @@ Cypress.Commands.add('getCurrentViewportSize', function () {
   }));
 });
 
-Cypress.Commands.add('showNavGroups', (open) => {
-  cy.get('body').then((body) => {
-    const isVisible = !!body.find('[data-testid=page-navigation]').length;
-    const isDialogOpen = !!body.find('[data-testid=page-navigation-dialog]').length;
+Cypress.Commands.add('showNavGroups', () => {
+  cy.findByRole('button', { name: 'Skjemasider' }).click();
+  cy.findByRole('dialog', { name: 'Skjemasider' }).should('be.visible');
+  cy.findByRole('dialog', { name: 'Skjemasider' }).should('have.css', 'opacity', '1');
+});
 
-    if (open && isVisible) {
-      return;
-    }
-    if (open && !isVisible) {
-      cy.findByRole('button', { name: 'Skjemasider' }).click();
-      cy.findByRole('dialog', { name: 'Skjemasider' }).should('be.visible');
-      cy.findByRole('dialog', { name: 'Skjemasider' }).should('have.css', 'opacity', '1');
-    }
-    if (!open && isDialogOpen) {
-      cy.findByRole('dialog', { name: 'Skjemasider' }).within(() => cy.findByRole('button', { name: 'Lukk' }).click());
-      cy.findByRole('dialog', { name: 'Skjemasider' }).should('not.exist');
-    }
-  });
+Cypress.Commands.add('hideNavGroups', () => {
+  cy.findByRole('dialog', { name: 'Skjemasider' }).within(() => cy.findByRole('button', { name: 'Lukk' }).click());
+  cy.findByRole('dialog', { name: 'Skjemasider' }).should('not.exist');
 });
 
 Cypress.Commands.add('navGroup', (groupName, pageName) => {
   if (pageName) {
-    cy.get('[data-testid=page-navigation]').then((container) =>
-      cy
-        .findByRole('button', { name: groupName, container })
-        .parent()
-        .then((container) => cy.findByRole('button', { name: pageName, container })),
-    );
+    cy.get('[data-testid=page-navigation]')
+      .findByRole('button', { name: groupName })
+      .parent()
+      .findByRole('button', { name: pageName });
   } else {
-    cy.get('[data-testid=page-navigation]').then((container) =>
-      cy.findByRole('button', { name: groupName, container }),
-    );
+    cy.get('[data-testid=page-navigation]').findByRole('button', { name: groupName });
   }
 });
 
-Cypress.Commands.add('gotoNavGroup', (groupName, pageName) => {
-  cy.showNavGroups(true);
-
-  cy.get('body').then((body) => {
-    const isUsingDialog = !!body.find('[data-testid=page-navigation-trigger]').length;
-
-    if (pageName) {
-      cy.navGroup(groupName).then((group) => {
-        if (group[0].getAttribute('aria-expanded') === 'false') {
-          cy.navGroup(groupName).click();
-        }
-      });
-      cy.navGroup(groupName).should('have.attr', 'aria-expanded', 'true');
-      cy.navGroup(groupName, pageName).click();
-      if (isUsingDialog) {
-        cy.findByRole('dialog', { name: 'Skjemasider' }).should('not.exist');
-      } else {
-        cy.navGroup(groupName, pageName).should('have.attr', 'aria-current', 'page');
+Cypress.Commands.add('gotoNavGroup', (isUsingDialog, groupName, pageName) => {
```

Comment:
It's only when detecting whether or not the dialog is currently open that caused problems right? So adding `isUsingDialog` as a prop here was not really needed? I just think it makes the command signature a bit messy, and I would like to have a nice utility command for navigating with the new navigation, similar to our `gotoNavPage` command for the old `NavigationBar` that gets used a lot in other tests

Reply:
>So adding `isUsingDialog` as a prop here was not really needed?

Maybe I'm not getting what you're saying, but it's needed as long as the behavior is supposed to be different depending on that prop. I guess we could try to read the current viewport size and infer if the dialog should be used from that? Or, which style/approach do you suggest?

Reply:
I was suggesting the original approach:
```ts
cy.get('body').then((body) => {
  const isUsingDialog = !!body.find('[data-testid=page-navigation-trigger]').length;
```
Since this should be static as long as the viewport is not changed during the test, as opposed to `isDialogOpen` where I can see that there is a problem.

Reply:
Duh, of course! Thanks! Sorry for being slow.. :sweat_smile: